Source/SwiftLintFramework/Rules/Lint/BlanketDisableCommandRule.swift

```diff
@@ -0,0 +1,143 @@
+import SwiftSyntax
+
+struct BlanketDisableCommandRule: ConfigurationProviderRule {
+    typealias ConfigurationType = BlanketDisableCommandConfiguration
+
+    var configuration = BlanketDisableCommandConfiguration()
+
+    init() {}
+
+    static let description = RuleDescription(
+        identifier: "blanket_disable_command",
+        name: "Blanket Disable Command",
+        description: "swiftlint:disable commands should be re-enabled before the end of the file",
+        kind: .lint,
+        nonTriggeringExamples: [
+            Example("""
+            // swiftlint:disable unused_import
+            // swiftlint:enable unused_import
+            """),
+            Example("""
+            // swiftlint:disable unused_import unused_declaration
+            // swiftlint:enable unused_import
+            // swiftlint:enable unused_declaration
+            """),
+            Example("// swiftlint:disable:this unused_import"),
+            Example("// swiftlint:disable:next unused_import"),
+            Example("// swiftlint:disable:previous unused_import")
+        ],
+        triggeringExamples: [
+            Example("// ↓swiftlint:disable unused_import"),
+            Example("""
+            // ↓swiftlint:disable unused_import unused_declaration
+            // swiftlint:enable unused_import
+            """),
+            Example("""
+            // swiftlint:disable unused_import
+            // ↓swiftlint:disable unused_import
+            // swiftlint:enable unused_import
+            """),
+            Example("""
+            // ↓swiftlint:enable unused_import
+            """)
+        ].skipWrappingInCommentTests().skipDisableCommandTests()
+     )
+
+    func validate(file: SwiftLintFile) -> [StyleViolation] {
+        var violations: [StyleViolation] = []
+        var ruleIdentifierToCommandMap: [RuleIdentifier: Command] = [:]
+        var disabledRuleIdentifiers: Set<RuleIdentifier> = []
+
+        for command in file.commands {
+            if command.action == .disable {
+                let alreadyDisabledRuleIdentifiers = command.ruleIdentifiers.intersection(disabledRuleIdentifiers)
+                violations.append(contentsOf: alreadyDisabledRuleIdentifiers.map {
+                    let reason = "The disabled '\($0.stringRepresentation)' rule was already disabled"
+                    return violation(forFile: file, command: command, reason: reason)
+                })
```

Comment:
It would help me understand the different checks better if this was a method call with an expressive name:
```suggestion
                violations.append(contentsOf: validateAlreadyDisabledRules(...))
```
Same for the other branches/checks. What do you think?

Reply:
Yep, I've made this change - definitely an improvement